draft-sharabayko-srt.md

```diff
@@ -1667,8 +1690,96 @@ of the higher bit of the difference.
+parties will generate identical cookies. However, this situation may occur if an
+application tries to "connect to itself" - that is, either connects to a local
+IP address, when the socket is bound to INADDR_ANY, or to the same IP address to
+which the socket was bound. If the cookies are identical (for any reason), the
+connection will not be made until new, unique cookies are generated (after a
+delay of up to one minute). In the case of an application "connecting to itself",
+the cookies will always be identical, and so the connection will never be established.
+
+If there is no response from a peer the WAVEAHAND handshake SHOULD be repeated every 250 ms
+until a connection timeout expires. The connection timeout value is defined by the implementation.
+
+If a WAVEAHAND of CONCLUSION handshake is received from the peer, the state is transitioned to the Attention.
+
+#### Conclusion {#conclusion-state}
+
+In the Conclusion state the party has received and now knows the peer's cookie value.
+Thus it can perform the Cookie Contest
+(compare both cookie values according to {{cookie-contest}})
+and resolve its role.
+The resolution of the Handshake Role (Initiator or Responder) is essential for further processing.
+
+Initiator replies with a Conclusion request handshake:
+
+- HS Version: 5
+- Extension field: appropriate flags.
+- Encryption field: advertised PBKEYLEN
+- Handshake Type: CONCLUSION.
+- Required Handshake Extension: HS Extension Message ({{handshake-extension-msg}})
+  with HS Extension Type SRT_CMD_HSREQ.
+- Other handshake extensions are allowed.
+
+If encryption is on, the Initiator will use either his
+own cipher family and block size or the one received from Alice (if she has advertised
+those values).
+
+The Responder with a Conclusion or a WAVEAHAND handshake without extensions until it receives the Conclusion Request from the peer:
+
+- HS Version: 5
+- Extension field: 0.
+- Encryption field: advertised PBKEYLEN.
+- Handshake Type: CONCLUSION.
+- Handshake extensions are NOT allowed.
+
+TODO: What to do if WAVEAHAND or AGREEMENT or else is received in this stage?
+Repeat conclusion response but not more often that every 250 ms.
+
+#### Initiated
+
+Alice receives Bob's CONCLUSION message. While at this point she also
+   performs the "cookie contest", the outcome will be the same. She switches to the
+   "fine" state, and sends:
+   - Version: 5
+   - Appropriate extension flags and encryption flags
+   - Handshake Type: CONCLUSION.
+
+   Both parties always send extension flags at this point, which will
+   contain HSREQ if the message comes from an Initiator, or
+   HSRSP if it comes from a Responder. If the Initiator has received a
+   previous message from the Responder containing an advertised cipher family and block size in the
+   encryption flags field, it will be used as the key length
+   for key generation sent next in the KMREQ extension.
```

Comment:
```suggestion
Alice receives Bob's CONCLUSION message. While at this point she also
   performs the "cookie contest" operation, the outcome will be the same. She switches to the
   "fine" state, and sends:
   - Version: 5
   - Appropriate extension flags and encryption flags
   - Handshake Type: CONCLUSION

   Both parties always send extension flags at this point, which will
   contain HSREQ if the message comes from an Initiator, or
   HSRSP if it comes from a Responder. If the Initiator has received a
   previous message from the Responder containing an advertised cipher family and block size in the
   encryption flags field, it will be used as the key length
   for key generation sent next in the KMREQ extension.
```